# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: modules/libjar/test/unit/test_bug333423.js
# Commit: 17a501a19918
# Full Hash: 17a501a1991842d8488ff3e7d4611bb688abfc87
# Author: Randell Jesup <rjesup@wgate.com>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1744043
# File Overlap Count: 1
# Description:
#   Bug 1744043: Clean up nsJAR r=nika,valentin,extension-reviewers,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132794
# ==============================================================================

diff -r daed1e9ef335 -r 17a501a19918 modules/libjar/test/unit/test_bug333423.js
--- a/modules/libjar/test/unit/test_bug333423.js	Thu Mar 24 13:45:28 2022 +0000
+++ b/modules/libjar/test/unit/test_bug333423.js	Thu Mar 24 13:54:56 2022 +0000
@@ -15,6 +15,10 @@
   );
   zipreader.open(file);
   zipreader.close();
-  var entries = zipreader.findEntries("*.*");
-  Assert.ok(!entries.hasMore()); // this shouldn't crash
+  // this should error out and not crash
+  Assert.throws(
+    () => zipreader.findEntries("*.*"),
+    /NS_ERROR_FAILURE/,
+    "Should error out on a closed zipreader"
+  );
 }